Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MultiPositionController.cs

```diff
@@ -0,0 +1,1001 @@
+                        }
+                    }
+                    if (Locomotive.DynamicBrakePercent == -1) Locomotive.SetDynamicBrakePercent(0);
+                    if (Locomotive.ThrottlePercent < 1 && Locomotive.DynamicBrakePercent < 100)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent + 2f);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeDecrease)
+                {
+                    if (Locomotive.DynamicBrakePercent > 0)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent - 1);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.Drive || controllerPosition == ControllerPosition.ThrottleHold)
+                {
+                    if (Locomotive.DynamicBrakePercent < 2)
+                    {
+                        Locomotive.SetDynamicBrakePercent(-1);
+                    }
+                    if (Locomotive.DynamicBrakePercent > 1)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent - 1);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.TrainBrakeIncrease)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() != "apply")
+                        {
+                            String test = Locomotive.TrainBrakeController.GetStatus().ToLower();
+                            Locomotive.StartTrainBrakeIncrease(null);
+                        }
+                        else
+                        {
+                            Locomotive.StopTrainBrakeIncrease();
+                        }
+                    }
+                }
+                else if (controllerPosition == ControllerPosition.Drive)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() != "release")
+                        {
+                            String boom = Locomotive.TrainBrakeController.GetStatus().ToString();
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        else
+                            Locomotive.StopTrainBrakeDecrease();
+                    }
+                }
+                if (controllerPosition == ControllerPosition.TrainBrakeDecrease)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() != "release")
+                        {
+                            String boom = Locomotive.TrainBrakeController.GetStatus().ToString();
```

Comment:
Code duplication, again the same comments

Reply:
replaced with enum comparison